src/crimson/os/seastore/transaction_manager.h

```diff
@@ -178,8 +178,14 @@ class TransactionManager : public ExtentCallbackInterface {
   {
     auto v = pin->get_logical_extent(t);
     if (v.has_child()) {
-      return v.get_child_fut().then([](auto extent) {
-	return extent->template cast<T>();
+      return v.get_child_fut(
+      ).then([this, pin = std::move(pin), &t](auto extent) mutable {
+        if (extent->is_fully_loaded()) {
```

Comment:
See the above comments, extent here should be always fully loaded if `Cache::get_extent_viewable_by_trans()` is patched correctly.